students/idcrickmore/lesson04/mailroom_pt2.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+from operator import itemgetter
+import os
+import datetime
+
+
+def menu():
+
+    print("""\n------------------ MENU ------------------\n
+PLEASE CHOOSE FROM THE FOLLOWING THREE OPTIONS
+1. Send Thank You
+2. Export Letters to Everyone
+3. Create a Report
+4. Quit\n""")
+
+    choice = input("-> ").lower()[0]
+
+    # use .get() on the 'options' dict to call functions
+    # keep calling the 'menu' function if user input not in dict
+    options.get(choice, menu)()
+
+
+def thankyou():
+
+    print("""\n------------------ THANK YOU ------------------\n
+type 'list' to to see a complete list of donors -
+type 'menu' at any time to return to the menu""")
+
+    # asks for a name from the user, or for the 'list' prompt
+    name_check = input("Enter first and last name of a donor\n\n-> ").lower()
+
+    if name_check not in options:
+
+        donation = float(input("What is the donation amount? -> "))
+
+        name_check = name_check.title()
+
+        # check to see if 'name_check' is in the donor dict...
+        # if it is, append it
+        if name_check in donors:
+            donors[name_check].append(donation)
+
+        # if 'name_check' isn't in the donor dict...
+        # set  the default format value as a list ([]) and append it
+        else:
+            donors.setdefault((name_check), []).append(donation)
+
+        letter_dict = {"name": name_check, "don": donation}
+
+        print("""\n------------------ Letter ------------------\n
+Dear {name},
+
+You rock. Your fat contribution of ${don:,.2f}
+will go a long way to lining my pockets.
+
+Sincerely,
+Scrooge McDuck""".format(**letter_dict))
```

Comment:
since you're not really using `letter_dict` anywhere else, you could do this inline:
 `.format(**{"name": name_check, "don": donation})`